### PR TITLE
Request module refactor

### PR DESCRIPTION
**Pull request by cfresquet**
Replacement for https://github.com/highfidelity/hifi/pull/10339, since Howard pointed out that I should create a branch in my fork, and use that to do the pull request from, instead of from my master.

Pulled out the "request" function from pal.js, makeUserConnection.js, and tablet-goto.js to a new request.js file in the modules directory.

Test: Confirm handshake functionality still works as expected.
Confirm connections/friends list & info still function as expected.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 28b166c51fd74201502a1af0622795817bd95331 available at: https://gist.github.com/91cc4607b98834d892ac182c3b18bc3c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-28b166c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-28b166c.dmg

File: scripts/system/makeUserConnection.js [hunk 1]
```diff
@@ -123,64 +126,13 @@
             [].map.call(arguments, JSON.stringify)));
     }
 
+    function debugPrint() {
```
Comment (howard-stearns) [inline scripts/system/makeUserConnection.js]: is this used? is it better than debug, just above? what is it for and why add it?

File: scripts/system/makeUserConnection.js [hunk 2]
```diff
@@ -11,7 +11,10 @@
 //  Distributed under the Apache License, Version 2.0.
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 //
-(function () { // BEGIN LOCAL_SCOPE
+
+(function() { // BEGIN LOCAL_SCOPE
```
Comment (howard-stearns) [inline scripts/system/makeUserConnection.js]: why the change?  linters/standard requires a space between the keyword function and the (in this case, missing/anonymou) function name.

Comment (cfresquet) [inline scripts/system/makeUserConnection.js]: Probably a cut/paste from somewhere else while I was editing.  It wasn't a conscious decision.  

I'll fix.

File: scripts/system/makeUserConnection.js (hunk 1 above)
Comment (cfresquet) [inline scripts/system/makeUserConnection.js]: It isn't used any more.  I'd overlooked removing it after I didn't need it.

(It was more generic than the debug, and simpler to use )

**Comment by hifi-gustavo** [comment]
Build console output for commit: afa72210e0bb594700323393d5d0d8ddf9d3f8d8 available at: https://gist.github.com/91cc4607b98834d892ac182c3b18bc3c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-afa7221.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-afa7221.dmg

**Comment by howard-stearns** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: afa72210e0bb594700323393d5d0d8ddf9d3f8d8 available at: https://gist.github.com/91cc4607b98834d892ac182c3b18bc3c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-afa7221.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-afa7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: afa72210e0bb594700323393d5d0d8ddf9d3f8d8 available at: https://gist.github.com/91cc4607b98834d892ac182c3b18bc3c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-afa7221.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10362-afa7221.dmg
